backend/danswer/background/celery/celery_utils.py

```diff
@@ -73,6 +85,11 @@ def should_prune_cc_pair(
             return True
         return False
 
+    if PREVENT_SIMULTANEOUS_PRUNING and most_recent_generic_pruning_task:
+        if check_live_task_not_timed_out(most_recent_generic_pruning_task, db_session):
```

Comment:
nit: while we're at it can we also make this name more descriptive. Right now it's not clear from the name what the behavior is if it's not live. E.g. something like `check_task_is_live_and_not_timed_out`